contrib/endpoints/src/api_manager/check_security_rules.cc

```diff
@@ -14,92 +14,350 @@
+            });
 }
 
-const std::string &AuthzChecker::GetAuthToken() {
-  // TODO: Get Auth token for accessing Firebase Rules service.
-  static std::string empty;
-  return empty;
+void AuthzChecker::FirebaseCheck(
+    std::string &ruleset_id, std::shared_ptr<context::RequestContext> context,
+    std::function<void(Status status)> continuation) {
+  auto checker = GetPtr();
+
+  std::string body;
+  Status status = BuildTestRequestBody(context, body);
+  if (!status.ok()) {
+    continuation(status);
+    return;
+  }
+
+  HttpFetch(AuthzChecker::GetRulesetTestUri(ruleset_id), "POST", body,
+            [context, continuation, checker, ruleset_id](Status status,
+                                                         std::string &&body) {
+
+              if (status.ok()) {
+                checker->env_->LogDebug(
+                    std::string("Test API succeeded with ") + body);
+                status = checker->ParseTestResponse(context, body);
+              } else {
+                checker->env_->LogError(std::string("Test API failed with ") +
+                                        status.ToString());
+                status = Status(Code::INTERNAL, kFailedFirebaseTest);
+              }
+
+              continuation(status);
+            });
+}
+
+Status AuthzChecker::ParseReleaseResponse(const std::string &json_str,
+                                          std::string *ruleset_id) {
+  grpc_json *json = grpc_json_parse_string_with_len(
+      const_cast<char *>(json_str.data()), json_str.length());
+
+  if (!json) {
+    return Status(Code::INVALID_ARGUMENT, kInvalidResponse);
+  }
+
+  Status status = Status::OK;
+  const char *id = GetStringValue(json, "rulesetName");
+  (*ruleset_id) = (id == nullptr) ? "" : GetStringValue(json, "rulesetName");
+
+  if (ruleset_id->empty()) {
+    env_->LogError("Empty ruleset Id received from firebase service");
+    status = Status(Code::INTERNAL, kInvalidResponse);
+  } else {
+    env_->LogInfo(std::string("Received ruleset Id: ") + *ruleset_id);
+  }
+
+  grpc_json_destroy(json);
+  return status;
+}
+
+Status AuthzChecker::ParseTestResponse(
+    std::shared_ptr<context::RequestContext> context,
+    const std::string &json_str) {
+  grpc_json *json = grpc_json_parse_string_with_len(
+      const_cast<char *>(json_str.data()), json_str.length());
+
+  if (!json) {
+    return Status(Code::INVALID_ARGUMENT,
+                  "Invalid JSON response from Firebase Service");
+  }
+
+  Status status = Status::OK;
+  Status invalid = Status(Code::INTERNAL, kInvalidResponse);
+
+  const grpc_json *testResults = GetProperty(json, "testResults");
+  if (testResults == nullptr) {
+    env_->LogError("TestResults are null");
+    status = invalid;
+  } else {
+    const char *result = GetStringValue(testResults->child, "state");
+    if (result == nullptr) {
+      env_->LogInfo("Result state is empty");
+      status = invalid;
+    } else if (std::string(result) != "SUCCESS") {
```

Comment:
I have declared a const char[] kTestSuccess with the rest of the cont strings. Let me know if this is not what you had in mind.